### PR TITLE
[#34] Make suggestions reload sleep configurable

### LINKED ISSUE #34: Make suggestions reload sleep configurable
Take a look here:

https://github.com/paypal/NNAnalytics/blob/a73dbc6a5a3dce96fdec027a28e91bdf6789cd1a/src/main/java/org/apache/hadoop/hdfs/server/namenode/NNLoader.java#L1949

We use a hardcoded 15 minute sleep time in NNA in between processing cached user reports.
It would be nice if this could be made configurable as part of the `security.properties`.

** We will eventually change the naming of the `security.properties` file to something like `nna.properties`.

### PR DESCRIPTION
**Pull request by pauljv92**
Make sure you have checked all steps below.

### GitHub Issue
Fixes #34 

### Checklist:

- [ ] This pull request updates the documentation
- [x] This pull request changes the code
- [x] Title of the PR is of format (example) : `[#25] Add Pull Request Template`, where `[#25] is the GitHub issue number`

### Commit Guidelines
- [x] My commits all reference GH issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

**Comment by pauljv92** [comment]
@zero45 changes made and validated!

File: src/main/java/org/apache/hadoop/hdfs/server/namenode/NNLoader.java [hunk 1]
```diff
@@ -2142,7 +2142,7 @@ public void initReloadThreads(ExecutorService internalService) {
               while (true) {
                 // Reload Keytab every 10 minutes.
                 try {
-                  Thread.sleep(10 * 60 * 1000L);
+                  Thread.sleep(conf.getSuggestionsReloadSleepTime());
```
Comment (zero45) [inline src/main/java/org/apache/hadoop/hdfs/server/namenode/NNLoader.java]: Let's not change the keytab reload time too. We can leave this hard-coded for now.

File: src/main/java/com/paypal/security/SecurityConfiguration.java [hunk 2]
```diff
@@ -101,6 +101,10 @@ public int getLdapConnectionPoolMaxSize() {
     return Integer.parseInt(properties.getProperty("ldap.connection.pool.max.size"));
   }
 
+  public int getSuggestionsReloadSleepTime() {
+    return Integer.parseInt(properties.getProperty("suggestions.reload.sleep.time"));
```
Comment (zero45) [inline src/main/java/com/paypal/security/SecurityConfiguration.java]: I've set a convention that anything that is not strictly security enabling parameters carry "nna." prepends.
I think we should do the same here -> "nna.suggestions.reload.sleep.time"

Preferably it should be -> "nna.suggestions.reload.sleep.ms"

**Comment by pauljv92** [comment]
@zero45 changes completed

File: src/main/java/org/apache/hadoop/hdfs/server/namenode/NNLoader.java [hunk 3]
```diff
@@ -2131,7 +2131,7 @@ public void initReloadThreads(ExecutorService internalService) {
                 }
                 // Reload suggestions every 15 minutes.
```
Comment (zero45) [inline src/main/java/org/apache/hadoop/hdfs/server/namenode/NNLoader.java]: This comment isn't necessary anymore.

**Pull request merged**